minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/DockerBuildTask.java

```diff
@@ -129,9 +142,48 @@ public void execDockerBuild() throws IOException, InterruptedException {
     List<String> execString = new ArrayList<>();
     execString.add(docker.get());
     execString.add("build");
+
+    if (!tag.isEmpty()) {
+      execString.add("-t");
+      execString.add(tag);
+    }
+
     execString.addAll(Arrays.asList(flags));
     execString.add(context);
 
     return execString;
   }
+
+  /**
+   * Builds the default tag for the built image in the form: {@code
+   * ${project.group}/${project.name}:${project.version}}.
+   *
+   * <p>If {@code ${project.name}}
+   *
+   * <p>If {@code ${project.group}} is empty, {@code ${project.group}/} will not be included.
+   *
+   * <p>If {@code ${project.version}} is empty, {@code :${project.version}} will not be included.
+   *
+   * <p>Replaces all instances of ':' in group, name, or version with '.'.
+   *
+   * @return the built tag
+   * @throws GradleException
+   */
+  // @VisibleForTesting
+  String buildDefaultTag() {
+    String group = getProject().getGroup().toString().replace(':', '.');
+    String projectName = getProject().getName().toString().replace(':', '.');
```

Comment:
I don't think gradle actually allows colons in projectName, are we expecting : somewhere?

Reply:
I wonder if replacing will be confusing, perhaps we just provide validation : "A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes" from https://docs.docker.com/engine/reference/commandline/tag/#description

Reply:
We can warn and set the tag to null if we can't generate a reasonable one.

Reply:
I tried putting a colon into a project name and it worked. I'll add the warning.

Reply:
Oh yeah, huh. Good call.